tools/codegen/gen.py

```diff
@@ -429,13 +484,48 @@ def generate_defn(faithful: bool) -> str:
     return at::_ops::{f.func.name.unambiguous_name()}::redispatch({exprs_str});
 }}
 """
+
         result = generate_defn(False)
         if sig_group.faithful_signature is not None:
             result += generate_defn(True)
 
         return result
 
 
+# Generates Static.cpp.
+# This generates static unboxing wrapper for ATen ops.
+@dataclass(frozen=True)
+class ComputeUnboxingWrapper:
+
+    @method_with_native_function
+    def __call__(self, f: NativeFunction) -> Optional[str]:
+        # We unconditionally generate function wrappers,
+        sig_group = CppSignatureGroup.from_native_function(f, method=False, fallback_binding=f.manual_cpp_binding)
+
+        def generate_defn(faithful: bool) -> str:
+            if faithful:
+                sig = sig_group.faithful_signature
+                assert sig is not None
+            else:
+                sig = sig_group.signature
+
+            # escape double quote in schema, get rid of extra double quotes
+            schema = json.dumps(sig.func.__str__())[1:-1]
+
+            return f"""
+OperatorGenerator(
```

Comment:
I also want to confirm - can we kill `register_c10_ops.cpp` as part of this PR? From my understanding after the initial conversation with @ljk53, we can use this codegen to directly register aten ops for torchscript, which is also what will get the unboxing codegen test coverage from server-side torchscript.

Reply:
I think we can, but there might be edge cases like ad-hoc `TORCH_LIBRARY` calls that are out of `native_functions.yaml`?

Reply:
I guess that means we can't get rid of `register_c10_ops.cpp` then, since the callback is necessary for custom ops?

It's not totally clear to me what will happen from this PR then in the jit registry on server-side. Technically, for a given aten ops, 2 registrations will happen:
- The codegen'd `OperatorGenerator()` call here will register the aten op to the jit registry
- The callback logic in `register_c10_ops.cpp` will **also** register the aten op to the jit registry, after it's been registered to the dispatcher?

Will double-registering be a problem for torchscript? Maybe we can fix that by modifying `register_c10_ops.cpp` to not register aten ops, and rely on the codegen to do it.

Also, let me know if you want to move the conversation to the new PR since I see this one is closed.

Reply:
Yeah seems like we are double registering for torchscript. I think adding a check makes sense, but that might be out of the scope for this PR? In the new PR #69881 I made codegen unboxing an option on `gen.py`. Maybe we can have another PR targeting for TorchScript and we can remove the option over there?

Reply:
Double-registering ops sounds like something that would be problematic for torchscript, right? If that's the case, then I think we'd want to deal with them together (separate PR's is fine, as long as they're landed together).